src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -216,20 +216,10 @@ private CookieOptions BuildCookieOptions()
             return cookieOptions;
         }
 
-        private async Task FinishResponseOnce()
-        {
-            if (!_finishCalled)
-            {
-                _finishCalled = true;
-                await FinishResponseAsync();
-                //await HandleAutomaticChallengeIfNeeded();
-            }
-        }
-
         protected virtual async Task FinishResponseAsync()
         {
             // Only renew if requested, and neither sign in or sign out was called
-            if (!_shouldRefresh || SignInAccepted || SignOutAccepted)
+            if (!_shouldRefresh || SignInCalled || SignOutCalled)
```

Comment:
Test hole here, need to add a test that verifies cookies aren't renewed if SignIn or SignOut is called. (this was broken before this change)